include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -100,6 +100,26 @@ class MGConstrainedDoFs : public Subscriptor
     const std::set<types::boundary_id> &boundary_ids,
     const ComponentMask &               component_mask = ComponentMask());
 
+  /**
+   * Fill the internal data structures with information
+   * about no normal flux boundary dofs.
+   *
+   * This function is limited to meshes whose no normal flux boundaries
+   * have faces which are normal to the x-, y-, or z-axis. Also, for a
+   * specific boundary id, all faces must be facing in the same direction,
+   * i.e., a boundary normal to the x-axis must have a different boundary
+   * id than a boundary normal to the y- or z-axis and so on. If the mesh
+   * here was produced using the GridGenerator class, setting
+   * <tt>colorize=true</tt> during mesh generation and calling
+   * make_no_normal_flux_constraints() for each no normal flux boundary is
+   * sufficient.
+   */
+  template <int dim, int spacedim>
+  void
+  make_no_normal_flux_constraints(const DoFHandler<dim, spacedim> &dof,
+                                  const types::boundary_id         bid,
+                                  const unsigned int first_vector_component);
+
```

Comment:
Should this be initialized as `const unsigned int first_vector_component = 0`?

Reply:
I'd say leave it explicit.